lib/services/local/integrations.go

```diff
@@ -173,6 +197,41 @@ func notReferencedByEAS(ctx context.Context, bk backend.Backend, name string) ([
 	return conditionalActions, nil
 }
 
+// integrationReferencedByAWSICPlugin returns an error if the integration name is referenced
+// by an existing AWS Identity Center plugin. In case the AWS Identity Center plugin exists
+// but does not reference this integration, a conditional action is returned with a revision
+// of the plugin to ensure that plugin hasn't changed during deletion of the AWS OIDC integration.
+// Returns nil ConditionalAction if AWS Identity Center plugin is not found.
+func integrationReferencedByAWSICPlugin(ctx context.Context, bk backend.Backend, name string) (*backend.ConditionalAction, error) {
+	pluginService := NewPluginsService(bk)
+	plugins, err := pluginService.GetPlugins(ctx, false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	for _, p := range plugins {
+		pluginV1, ok := p.(*types.PluginV1)
+		if !ok {
+			continue
+		}
+
+		if pluginV1.GetType() == types.PluginType(types.PluginTypeAWSIdentityCenter) {
+			switch pluginV1.Spec.GetAwsIc().IntegrationName {
+			case name:
+				return nil, trace.BadParameter("cannot delete AWS OIDC integration currently referenced by AWS Identity Center integration")
+			default:
+				return &backend.ConditionalAction{
+					Key:       backend.NewKey(pluginsPrefix, name),
+					Action:    backend.Nop(),
+					Condition: backend.Revision(pluginV1.GetRevision()),
+				}, nil
+			}
+		}
+	}
+
+	return nil, nil
```

Comment:
Returning `nil, nil` is an anti-pattern, especially if the first argument is a pointer - because callers usually expect the first argument to be usable if the err is nil so that's a potential panic. If you change the function to return a slice as per my suggestion above, that would be fine.

Reply:
Updated to return slice https://github.com/gravitational/teleport/pull/48052/commits/ca33f8a76315374bbf1c2ad0cf87249164b5b2ce#diff-dafac049a7b024e6676485f78dc968156931955542c16fc743deeff59def2c25R202